### PR TITLE
added solutions and testing to task 8 - PATCH /api/articles/:article_id

### PR DESCRIPTION
**Pull request by JordanEckford**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -227,5 +227,67 @@ describe("POST /api/articles/:article_id/comments", () => {
...
+   .then(({ body }) => {
+    expect(body.article).toEqual(
+     expect.objectContaining({
+      article_id: 3,
+      title: "Eight pug gifs that remind me of mitch",
+      topic: "mitch",
+      author: "icellusedkars",
+      body: "some gifs",
+      created_at: expect.any(String),
+      votes: 20,
```
Comment (roisinoreilly) [inline __tests__/app.test.js:246]: this test would be better if you checked what the votes were before the patch request and then after. While it's good to check that all the other properties on the object are still there, I think it's maybe not needed if we're just concerned about the votes

File: __tests__/app.test.js [hunk 2]
```diff
@@ -227,5 +227,67 @@ describe("POST /api/articles/:article_id/comments", () => {
...
+      article_img_url:
+       "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+     })
+    );
+   });
+ });
+ test("should return the correct status and response when passed an incorrect formatted article_id", () => {
```
Comment (roisinoreilly) [inline __tests__/app.test.js:253]: I would specify that it should return a 400/404 etc in your test description

File: __tests__/app.test.js [hunk 3]
```diff
@@ -227,5 +227,67 @@ describe("POST /api/articles/:article_id/comments", () => {
...
+  const testObject = { bananas: 20 };
+  return request(app)
+   .patch("/api/articles/3")
+   .send(testObject)
+   .expect(400)
+   .then(({ body }) => {
+    expect(body.msg).toBe("post request incorrect");
+   });
+ });
+ test("should respond with appropriate error when object with correct property with additional properties is sent to valid article_id", () => {
```
Comment (roisinoreilly) [inline __tests__/app.test.js:283]: these last two edge cases are excellent, great job

File: controllers/articles.controller.js [hunk 4]
```diff
@@ -16,3 +20,19 @@ exports.getArticles = (req, res, next) => {
   res.status(200).send({ articles });
  });
 };
+
+exports.patchArticleByID = (req, res, next) => {
+ const { article_id } = req.params;
+ const { inc_votes } = req.body;
+
+ if (inc_votes === undefined || Object.keys(req.body).length > 1) {
```
Comment (roisinoreilly) [inline controllers/articles.controller.js:28]: really nicely handled

**Review by roisinoreilly — APPROVED** [review]
really great work on this ticket :) the only change I'd like you to implement before merging is to check what the votes are before the patch as well as after. Once you've done that PR is approved :)
